src/transport/SecureSessionMgr.h

```diff
@@ -91,18 +91,18 @@ class SecureSessionHandle
 class EncryptedPacketBufferHandle final : private System::PacketBufferHandle
 {
 public:
-    EncryptedPacketBufferHandle() : mMsgId(0) {}
-    EncryptedPacketBufferHandle(EncryptedPacketBufferHandle && aBuffer) :
-        PacketBufferHandle(std::move(aBuffer)), mMsgId(aBuffer.mMsgId)
-    {}
+    EncryptedPacketBufferHandle() {}
+    EncryptedPacketBufferHandle(EncryptedPacketBufferHandle && aBuffer) : PacketBufferHandle(std::move(aBuffer)) {}
+
+    void operator=(EncryptedPacketBufferHandle && aBuffer) { PacketBufferHandle::operator=(std::move(aBuffer)); }
 
-    void operator=(EncryptedPacketBufferHandle && aBuffer)
+    uint32_t GetMsgId() const
     {
-        PacketBufferHandle::operator=(std::move(aBuffer));
-        mMsgId                      = aBuffer.mMsgId;
-    }
+        PacketHeader header;
+        uint16_t headerSize = 0;
 
-    uint32_t GetMsgId() const { return mMsgId; }
+        return (header.Decode((*this)->Start(), (*this)->DataLength(), &headerSize) == CHIP_NO_ERROR) ? header.GetMessageId() : 0;
```

Comment:
consider adding a log for decoding errors. A bug here would be very difficult to track otherwise.

Reply:
Acked